actuator.c

```diff
@@ -1,94 +1,44 @@
-        RELAY_OFF();
-    } else if (SAFE_STATE == ACTUATOR_ON) {
-        RELAY_ON();
-    } else {
-        // Should not get here.
-        while (1) {}; // panic, this will be caught early on in testing
-    }
-    
-#ifdef INJECTOR
-    LATB4 = 1;
-    TRISB4 = 0; // set LIMIT_OPEN as output to power hall sensor
-
-    TRISB3 = 1; // set LIMIT_CLOSED (pin 24) as input
-    ANSELB3 = 1; // set as analog input
-#else
-    TRISB4 = 1; // set LIMIT_OPEN (pin 25) as input
-    ANSELB4 = 0; // set as digital input
+uint8_t actuator_states = 0;
 
-    TRISB3 = 1; // set LIMIT_CLOSED (pin 24) as input
-    ANSELB3 = 0; // set as digital input
-#endif
+void actuator_init() {
+    actuator_states = pca_get_output();
 }
 
-void actuator_set(enum ACTUATOR_STATE state) {
+void actuator_set(enum ACTUATOR_STATE state, uint8_t pin_num) {
     if (state == ACTUATOR_OFF) {
-        WHITE_LED_OFF();
-        RELAY_OFF();
+        actuator_states &= ~(1 << pin_num);
     } else if (state == ACTUATOR_ON) {
-        WHITE_LED_ON();
-        RELAY_ON();
+        actuator_states |= (1 << pin_num);
     }
+    pca_set_output(actuator_states);
 }
 
-enum ACTUATOR_STATE get_actuator_state(void) {
-#if !HAS_LIMS
-    return ACTUATOR_UNK;
-#else
-// define in board.h
-#ifdef INJECTOR
-    adc_result_t hall_raw = ADCC_GetSingleConversion(channel_HALL);
-    
-    if (hall_raw > HALL_ERR_THRESHOLD) { return ACTUATOR_ILLEGAL; }
-    // HIGH_STATE is defined in board.h as the ACTUATOR_STATE that a high
-    // hall sensor reading maps to.
-    if (hall_raw > HALL_THRESHOLD) { return HIGH_STATE; }
-    return 1 - HIGH_STATE; // First two ACTUATOR_STATEs are on and off so this works to invert
-#else
-    // read limit switch values
-    bool actuator_open = PORTBbits.RB4;
-    bool actuator_closed = PORTBbits.RB3;
-    
-    if (actuator_open && !actuator_closed) { return ACTUATOR_OFF; }
-    if (!actuator_open && actuator_closed) { return ACTUATOR_ON; }
-    if (!actuator_open && !actuator_closed) { return ACTUATOR_UNK; }
-    return ACTUATOR_ILLEGAL; // both limit switches at same time
-#endif
-#endif
+void set_actuator_LED(enum ACTUATOR_STATE state, enum ACTUATOR_ID actuator) {
+    if (actuator == ACTUATOR_INJECTOR_VALVE) {
+        (state == ACTUATOR_ON) ? LED_ON_R() : LED_OFF_R();
+    } else { // Vent/Fill
+        (state == ACTUATOR_ON) ? LED_ON_B() : LED_OFF_B();
+    }
 }
 
-void actuator_send_status(enum ACTUATOR_STATE req_state) {
-// define in board.h
-#ifdef INJECTOR
-    adc_result_t hall_raw = ADCC_GetSingleConversion(channel_HALL);
-    
-    // Send a CAN message with the raw hall value
-    can_msg_t hall_msg;
-    build_analog_data_msg(millis(), SENSOR_MAG_1, hall_raw, &hall_msg);
-    txb_enqueue(&hall_msg);
-#endif
-    enum ACTUATOR_STATE curr_state = get_actuator_state();
+// TODO update this
+enum ACTUATOR_STATE get_actuator_state(uint8_t pin_num) {
+
+    // temp code
+    uint8_t state = (actuator_states >> pin_num) & 0x01;
+
+    if (state == 1) {
```

Comment:
we could use ternary operator instead of this iff condition

`return (state == 1) ? ACTUATOR_ON : ACTUATOR_OFF`